Cleaning up Mesos from paasta readthedocs - PAASTA-18313 (#3954)
Yelp/paasta: docs/source/about/glossary.rst

Glossary
========

**Docker**
~~~~~~~~~~

Container `technology <https://www.docker.com/whatisdocker/>`_ that
PaaSTA uses.

**Kubernetes**
~~~~~~~~~~~~~~

`Kubernetes <https://kubernetes.io/>`_ (a.k.a. k8s) is the open-source system on which Yelp runs many compute workloads.
In Kubernetes, tasks are distributed to and run by servers called Kubelets (but a.k.a. kube nodes or Kubernetes agents) from the Kubernetes control plane.

**Kubernetes Node**
~~~~~~~~~~~~~~~~~~~

A node is a worker machine in a Kubernetes cluster that runs Pods.
In our case, it's usually a virtual machine provisioned via AWS EC2 Fleets or AutoScalingGroups

**Kubernetes Horizontal Pod Autoscaler (HPA)**
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> It's a Kubernetes feature that automatically scales the number of pods in a deployment based on observed CPU utilization (or, with custom metrics support, on some other application-provided metrics).

Reviewer: to be consistent with the other definitions:
```suggestion
A Kubernetes feature that automatically scales the number of pods in a deployment based on observed CPU utilization (or, with custom metrics support, on some other application-provided metrics).
```

**clustername**
~~~~~~~~~~~~~~~

A shortname used to describe a PaaSTA cluster. Use \`paasta
list-clusters\` to see them all.

**Kubernetes pod**
~~~~~~~~~~~~~~~~~~~

Atomic deployment unit for PaaSTA workloads at Yelp and all Kubernetes clusters. Can be thought of as a collection of 1 or more related containers.
Pods can be seen as one or more containers that share a network namespace, at Yelp these are individual instances of one of our services, many can run on each server.

**instancename**
~~~~~~~~~~~~~~~~

> Logical collection of Kubernetes pods that comprise a Kubernetes Deployment. service

Reviewer: maybe something like:
```suggestion
Logical collection of Kubernetes pods that comprise an application deployed on Kubernetes. service
```
would be simpler for folks that don't know what a pod is? we haven't really defined what a Deployment is yet in k8s in this glossary otherwise :p

Author: That's a good idea, will add also what a deployment is to the glossary :p

> name + instancename = Kubernetes Deployment. Examples: main, canary. Each instance represents a running

Reviewer: ...although - maybe we should just define what a Deployment is so that this makes sense then!

version of a service with its own configuration and resources.

**namespace**
~~~~~~~~~~~~~

An haproxy/SmartStack concept grouping backends that listen on a

> particular port. A namespace may route to many healthy paaSTA

Reviewer: ```suggestion
particular port. A namespace may route to many healthy PaaSTA
```

> instances. By default, the namespace in which a Kubernetes deployment appears is

Author: I wonder if we should just say "paasta instance" since also an application is not defined here

> instances. By default, the namespace in which a Kubernetes deployment appears is

Author: will define as well kubernetes namespace up in the glossary
